app/utils/lava_log_parser.py

```diff
@@ -87,11 +114,19 @@ def run(log, boot, txt, html):
         timestamp = "<span class=\"timestamp\">{}  </span>".format(raw_ts)
 
         fmt = formats.get(level)
+        
+        if(isinstance(msg, list)):
+            msg = ' '.join(msg)  
+
         if fmt:
             log_buffer.append(timestamp + fmt.format(cgi.escape(msg)))
             numbers[level] += 1
-        elif level == "target":
-            log_buffer.append(timestamp + cgi.escape(msg) + "\n")
+        elif level == "target":            
+            if (re.search('\<([0-7])\>', msg)):
```

Comment:
This part of the logic needs to be rewritten:
* compile the regex in the module and use `.match()`
* add `^` at the beginning of the regex to only detect lines that start with `<>`
* use a list for the kernel log levels (see previous comment)